File: ts-cw-metrics/services/hook.ts
import { SNSEvent } from "aws-lambda";

const webHookUrl = '';

async function handler(event: SNSEvent) {
    for (const record of event.Records) {
        await fetch(webHookUrl, {
            method: 'POST',
            body: JSON.stringify({
                "text": `Huston, we have a problem: ${record.Sns.Message}`
            })
        })
    }
}


export { handler }

File: py-cw-metrics/services/hook.py
import urllib3
import json
http = urllib3.PoolManager()

def handler(event, context):
    print('calling slack!!!')
    url = ""
    msg = {
        "channel": "#aws-events",
        "text": event['Records'][0]['Sns']['Message'],
    }
    
    encoded_msg = json.dumps(msg).encode('utf-8')
    resp = http.request('POST',url, body=encoded_msg)
    print({
        "message": event['Records'][0]['Sns']['Message'], 
        "status_code": resp.status, 
        "response": resp.data
    })